frontend/src/studio/components/drum-machine/DrumMachine.tsx

```
@@ -549,74 +571,61 @@ const DrumMachine: React.FC<DrumMachineProps> = ({
+      await removeSamplerTrack(samplerTrackIdToDelete);
+      console.log(`DrumMachine (${trackId}): removeSamplerTrack store action finished.`);
+      
+      // THEN update local state (keep existing logic from reset file)
+      // This part assumes the reset file had this logic based on index
+      const newDrumNames = [...drumNames]; newDrumNames.splice(index, 1); setDrumNames(newDrumNames);
+      const newGrid = [...grid]; newGrid.splice(index, 1); setGrid(newGrid);
+      const newDrumNotes = [...drumNotes]; newDrumNotes.splice(index, 1); setDrumNotes(newDrumNotes);
+      if (openPianoRolls.has(index)) closeInternalPianoRoll(index); // Keep if reset had internal rolls
+      const newOpenPianoRolls = new Set<number>(); openPianoRolls.forEach(rollIndex => { if (rollIndex < index) newOpenPianoRolls.add(rollIndex); else if (rollIndex > index) newOpenPianoRolls.add(rollIndex - 1); }); setOpenPianoRolls(newOpenPianoRolls);
+      if (selectedSound >= newDrumNames.length) setSelectedSound(newDrumNames.length - 1); else if (selectedSound === index) setSelectedSound(Math.min(index, newDrumNames.length - 1));
+      // Resizing will be handled by the useEffect watching samplerTrackIds
```

Biome (1.9.4)

[error] This code is unreachable



(lint/correctness/noUnreachable)